src/normalizers/genericnormalizer.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/normalizers/genericnormalizer
+ */
+
+import { unwrapParagraph, moveNestedListToListItem } from '../filters/common';
+import UpcastWriter from '@ckeditor/ckeditor5-engine/src/view/upcastwriter';
+
+/**
+ * Normalizer for the content pasted from different sources, where we can detect wrong syntax.
+ *
+ * @implements module:paste-from-office/normalizer~Normalizer
+ */
+export default class GenericNormalizer {
```

Comment:
Here are the details and comments about why it's important:
https://github.com/ckeditor/ckeditor5-paste-from-office/pull/72#issuecomment-517253224

Reply:
Does it happen only on Safari? If so I'd still ad some env checking to be sure that this normalizer doesn't fire up too often. @mlewand WDYT? OTOH this normalizer will always fix pasted lists (even from other sources) so it may be beneficial.

I see two routes:
1. Make it run only for Safari and name it more specific, like: `SafariListNormalizer`
2. Make it general list fixer - the I'd name it `NestedListNormalizer` and maybe add a check if pasted content has broken list (AFAICS `ul></ul`, `/p></li` and `/li><ul` could be checked - with `ul|ol` variants check) - but the check idea is  something to give another though if it makes sense.

Reply:
I would not stick to "List" name here. As content is unrecognizable for any filter.
Google Docs for some specific contents (with tables) under Safari doesn't provide any string which could provide us info, that content came from Google Docs. It's just regular broken HTML.
So the same situation will repeat for heading filters or any other further filters. Where we still won't be knowing that content came from Google Docs.

Reply:
So in such cases better could be `SafariNormalizer` however this might be a little bit confusing, as there will be content from Safari that sometimes will activate `SafariNormalizer` and sometimes `GoogleDocsNormalizer`
We can prioritize it, but then it will be an exact copy of GoogleDocs normalizer and in such case, in my opinion, better will be making a change in `isActive()` method of google normalizer:
```
isActive( htmlString ) {
	return googleDocsMatch.test( htmlString ) || env.isSafari;
}
```

Reply:
@jodator @msamsel I took a peek at it and I see that:

* Safari problem described in https://github.com/ckeditor/ckeditor5-paste-from-office/pull/72#issue-302421316 happens when table is first, but does not happen if there's anything preceding it, which makes the bug that much less common (safari * edge case = very rare case).
	
	First of all it does look like upstream issue, I suspected Safari at first, but given that id appears if table is not the first item it sounds it's more of a Google Docs bug. Have we checked if this issue can be reported and tracked?
* I don't feel good with exposing this filter as a generic feature if we don't have any need for it (no requests, no real use cases).

Since still pretty much all the cases will be handled on safari except this particular oddity, let's wait for some real life cases where it will be useful. Until then let's extract mentioned Safari problem as a separate issue and put it to backlog.

Reply:
@jodator I removed this normalizer.

Reply:
I extracted this case to separate issue: #75